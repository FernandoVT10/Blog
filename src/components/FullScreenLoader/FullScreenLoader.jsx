import "./FullScreenLoader.scss";

export default ({ loading, text = "Loading..." }) => {
    const fullScreenLoaderClass = loading ? "full-screen-loader--active" : "";
    
    return (
        <div className={`full-screen-loader ${fullScreenLoaderClass}`}>
            <div
            className="full-screen-loader__spinner spinner-border text-light"
            role="status">
                <span className="sr-only">Loading...</span>
            </div>

            <h3 className="full-screen-loader__text">{ text }</h3>
        </div>
    );
};