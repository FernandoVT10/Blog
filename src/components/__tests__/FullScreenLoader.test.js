import { create } from "react-test-renderer";
import FullScreenLoader from "../FullScreenLoader/";

test("Check if <FullScreenLoader/> renders correctly", () => {
    const component = create(<FullScreenLoader loading={true} text="Test text" />);

    expect(component.toJSON()).toMatchSnapshot();
});