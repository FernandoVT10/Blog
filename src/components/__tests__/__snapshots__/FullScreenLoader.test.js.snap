// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <FullScreenLoader/> renders correctly 1`] = `
<div
  className="full-screen-loader full-screen-loader--active"
>
  <div
    className="full-screen-loader__spinner spinner-border text-light"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <h3
    className="full-screen-loader__text"
  >
    Test text
  </h3>
</div>
`;
